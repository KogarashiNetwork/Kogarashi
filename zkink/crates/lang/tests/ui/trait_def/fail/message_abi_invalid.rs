use ink_lang as ink;

#[ink::trait_definition]
pub trait TraitDefinition {
    #[ink(message)]
    extern "C" fn message_ref(&self);

    #[ink(message)]
    extern "C" fn message_mut(&mut self);
}

fn main() {}
