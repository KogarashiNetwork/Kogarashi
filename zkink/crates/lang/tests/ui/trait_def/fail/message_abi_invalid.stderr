error: ink! trait methods with non default ABI are not supported
 --> tests/ui/trait_def/fail/message_abi_invalid.rs:6:5
  |
6 |     extern "C" fn message_ref(&self);
  |     ^^^^^^^^^^
